cli/src/send_tpu.rs

```diff
@@ -0,0 +1,31 @@
+use log::*;
+use solana_client::{rpc_client::RpcClient, rpc_response::RpcLeaderSchedule};
+use std::net::{SocketAddr, UdpSocket};
+
+pub fn get_leader_tpu(
+    rpc_client: &RpcClient,
+    slot_index: u64,
+    leader_schedule: Option<&RpcLeaderSchedule>,
+) -> Option<SocketAddr> {
+    let leader_schedule = leader_schedule?;
+    leader_schedule
+        .iter()
+        .find(|(_pubkey, slots)| slots.iter().any(|slot| *slot as u64 == slot_index))
+        .and_then(|(pubkey, _)| {
+            let cluster_nodes = rpc_client.get_cluster_nodes().ok()?;
```

Comment:
Let's pass `cluster_nodes` in as an arg to avoid the RPC call on every call.   Make the caller do it, once, outside of the loop of `send_and_confirm_transactions_with_spinner()`